realm/src/main/java/io/realm/Realm.java

```diff
@@ -114,12 +114,15 @@
     private static final String APT_NOT_EXECUTED_MESSAGE = "Annotation processor may not have been executed.";
     private static final String INCORRECT_THREAD_MESSAGE = "Realm access from incorrect thread. Realm objects can only be accessed on the thread they where created.";
     private static final String CLOSED_REALM = "This Realm instance has already been closed, making it unusable.";
+    private static final String INVALID_KEY_MESSAGE = "The provided key is illegal. It should either be null or be 64 bytes long";
+    private static final String DIFFERENT_KEY_MESSAGE = "Realm access with incorrect key";
 
     @SuppressWarnings("UnusedDeclaration")
     private static SharedGroup.Durability defaultDurability = SharedGroup.Durability.FULL;
     private boolean autoRefresh;
     private Handler handler;
 
+    private final byte[] key;
```

Comment:
It is a nitpick, but this implies that the key is present in memory. If an attacker has access to the phone at this point he can read the key directly. He got access to the database anyway at this point, but having the key would grant future access as well. 

I would probably hash this value using SHA-256 or something. It doesn't mean you cannot intercept the key anyway, but It would make it a little harder.